crates/core/src/client/config.rs

```diff
@@ -4,31 +4,64 @@ use starknet::providers::jsonrpc::{HttpTransport, JsonRpcTransport};
 use starknet::providers::JsonRpcClient;
 use url::Url;
 
+use super::constants::{KATANA_RPC_URL, MADARA_RPC_URL};
 use super::errors::ConfigError;
 
 fn get_env_var(name: &str) -> Result<String, ConfigError> {
     std::env::var(name).map_err(|_| ConfigError::EnvironmentVariableMissing(name.into()))
 }
 
-#[derive(Default)]
+#[derive(Default, Clone)]
+pub enum Network {
+    #[default]
+    Katana,
+    Madara,
+    Mainnet,
+    Goerli1,
+    Goerli2,
+    Mock,
+}
+
+#[derive(Default, Clone)]
 /// Configuration for the Starknet RPC client.
 pub struct StarknetConfig {
-    /// Rpc url.
-    pub url: String,
+    /// Rpc url. Optional as it can be provided by the environment variable `STARKNET_NETWORK`.
+    pub url: Option<Url>,
     /// Kakarot contract address.
     pub kakarot_address: FieldElement,
     /// Proxy account class hash.
     pub proxy_account_class_hash: FieldElement,
+    /// Additional configuration if the underlying provider is a Sequencer provider.
+    pub network: Network,
 }
 
 impl StarknetConfig {
-    pub fn new(url: String, kakarot_address: FieldElement, proxy_account_class_hash: FieldElement) -> Self {
-        StarknetConfig { url, kakarot_address, proxy_account_class_hash }
+    pub fn new(
+        url: Option<Url>,
+        kakarot_address: FieldElement,
+        proxy_account_class_hash: FieldElement,
+        network: Network,
+    ) -> Self {
+        StarknetConfig { url, kakarot_address, proxy_account_class_hash, network }
     }
 
     /// Create a new `StarknetConfig` from environment variables.
     pub fn from_env() -> Result<Self, ConfigError> {
-        let starknet_rpc = get_env_var("STARKNET_RPC_URL")?;
+        let starknet_rpc = get_env_var("STARKNET_NETWORK")?;
```

Comment:
I don't understand this restriction to a list of target. I think that the config should be instead

```
{
    pub provider_url: "mainnet" | "testnet" | "testnet2" | Url,
    pub kakarot_address: FieldElement,
    pub proxy_account_class_hash: FieldElement,
}
```

In the `new` we can match against the provider_url and instantiate a sequencer or rpc provider accordingly

In the `from_env` we just take this rpc from end and call `new`

Reply:
Alright, I'll try like this, 
The way to do this (union types) is through an enum! It's actually a great idea, thanks for the tip

Reply:
proposed a fix